v0.3/tutorials/quickstart counter solution.eve

```diff
@@ -0,0 +1,42 @@
+---
+layout: default
+title: Quickstart + Improved Counter
+---
+
+{% raw %}
+
+# Quickstart + Improved Counter
+
+This guide expands on the quickstart `Counter` page and shows you how to display the counter message on the button itself. The following code is meant to replace all the code from the original counter.
+
+## The self-contained counter
+
+````
+```
+search
+  [#app page: "Counter"]
+  view = [#qs-contents]
+  how-many = if n = gather/count[for: [#clicked]] then n else 0
+
+bind
+  view.children += [#ui/button #qs-increment text: "The button has been clicked {{how-many}} times" sort: 1]
+```
+````
+
+A common trope is that good programmers delete code, and this follows suit - we actually get a more elegant solution here by condensing two blocks before down into one. We can't just smash together the two blocks entirely though because we want the original message to read "The button has been clicked 0 times" before any clicks happen. Recall that in the original counter example, before any clicks occurred, the counter message wouldn't show up because the search was trying to `gather/count` a non-existent record and causing the whole block to fail. If we tried that here, the counter button wouldn't show up. To account for this, an `if` statement sets `how-many` to the number of clicks once they happen, and `else` handles the edge case of starting at 0.
```

Comment:
"A common trope is that good programmers delete code, and this follows suit" we don't have to make a value judgement here